lib/utils/process-jobs.ts

```diff
@@ -94,19 +97,26 @@ export const processJobs = function(this: Agenda, extraJob: Job) {
       return;
     }
 
-    // Set that we are running this
-    self._isLockingOnTheFly = true;
-
     // Grab a job that needs to be locked
     const now = new Date();
-    // @todo: We should check if job exists
-    const job = self._jobsToLock.pop()!;
+    const job = self._jobsToLock.pop();
+    if(job === undefined) {
+      debug('no job was popped from _jobsToLock, extremly unlikely but not impossible concurrency issue');
+      self._isLockingOnTheFly = false;
+      return;
+    }
+
+    if(self._isJobQueueFilling.has(job.attrs.name)) {
```

Comment:
i dont really understand the isJobQueueFilling check here.. if isLockingOnTheFly is true, it shouldn't come here anyway? Which means: there is only one "running" lockOnTheFly.
Or why/how can this flag be different to isLockingOnTheFly?